authentication/client/src/components/Login.tsx

```diff
@@ -0,0 +1,56 @@
+import { gql, useLazyQuery } from "@apollo/client";
+import { Button, Grid, Input, Typography } from "@mui/material";
+import React, { useEffect, useState } from "react";
+
+const GET_LOGIN = gql`
+  query ($name: String!, $password: String!) {
+    authenticate(name: $name, password: $password)
+  }
+`;
+
+interface Props {
+  setToken: (arg: string) => void;
+}
+
+export default function Login(props: Props) {
+  const [name, setName] = useState<String>("");
+  const [password, setPassword] = useState<String>("");
+
+  const [getlogin, { error, data }] = useLazyQuery(GET_LOGIN, {
```

Comment:
lets use loading state as well, the state sequence should be `loading ? spinner : (error ? errorMessage ? data)`